pilot/pkg/kube/inject/inject.go

```diff
@@ -570,7 +618,7 @@ func intoObject(sidecarTemplate string, meshconfig *meshconfig.MeshConfig, in ru
 	if metadata.Annotations == nil {
 		metadata.Annotations = make(map[string]string)
 	}
-	metadata.Annotations[sidecarAnnotationStatusKey] = status
+	annotationStatus.setValue(metadata.Annotations, status)
```

Comment:
e.g. here it looks like you're setting value in annotationStatus whereas you could do
metadata.Annotations[annotationStatus.Name] = status.

Reply:
ack ... reverted to setting the value on the annotations map inline.